src/modules/returns/lib/jobs/validate-returns.js

```diff
@@ -0,0 +1,74 @@
+const messageQueue = require('../../../../lib/message-queue');
+const JOB_NAME = 'validate-uploaded-returns-data';
+const returnsUpload = require('../../lib/returns-upload');
+const eventsService = require('../../../../lib/services/events');
+const errorEvent = require('./error-event');
+const { logger } = require('../../../../logger');
+const uploadValidator = require('../../lib/returns-upload-validator');
+const { mapMultipleReturn } = require('../../lib/upload-preview-mapper');
+const { set } = require('lodash');
+
+/**
+ * Validates the submitted returns data after it's been mapped to JSON
+ *
+ * @param {Object} data containing eventId and companyId
+ * @returns {Promise}
+ */
+const publishValidateReturnsStart = data =>
+  messageQueue.publish(JOB_NAME, returnsUpload.buildJobData(data));
+
+/**
+* Handler for the 'validate-uploaded-returns-data' job in PG Boss.
+*
+* This will acquire the saved document from AWS S3,
+* validate the JSON representation, then put the JSON
+* back to S3 for future processing.
+*
+* @param {Object} job The job data from PG Boss
+*/
+const handleValidateReturnsStart = async job => {
+  const { eventId, companyId } = job.data;
+  const event = await eventsService.findOne(eventId);
+  const jsonData = await loadJson(eventId);
```

Comment:
I think this could move inside the try { } block, then if there's a problem the event will be marked in error by the catch { }